res-opt.mod

```diff
@@ -0,0 +1,64 @@
+#param n
+#PoA(p1...pn)
+set R;
+#robot
+set r;
+#switches 
+set w ;
+#servers
+set s;
+#hardware graph V(G) nodes (i.e. PoA,robot,switches,servers)
+set N := {R union r union w union s};
+# graph edges of hardware graph
+set E:={n1 in N, n2 in N}; 
+# service graph
+set S; 
+# vertices (VNFs) of service graph
+set ES:={v1 in S,v2 in S}; 
+#set of VNFs
+set VNF{S}; 
+#total computational resources
+param comput_res{n in N};	
+#VNF computational resources
+param c { v in S};  

```

Comment:
A VNF is not within the set of services, but within the set of VNFs of a particular service, therefore, it should be something like:
```ampl
param c {Si in S, v in VNFs[Si] };
```